src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,19 @@
+package duke.task;
+
+public class Deadline extends Task {
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public String getBy() {
```

Comment:
maybe rename your by variable to dueDate? so your getter will be getDueDate, instead of getBy because it sounds rather ambiguous.